### PR TITLE
Add clustering feature reduction to a point feature layer fixes

### PR DESCRIPTION
**Pull request by williambohrmann3**
# Description

- Replaces sliders with pickers to avoid flickering
- Updates UI
- Fixes a crash caused by drawing clusters before map finishes loading

## Type of change

- Bug fix
- Other enhancement

## Platforms tested on

- [x] WPF .NET 8
- [x] WPF Framework
- [x] WinUI
- [x] MAUI WinUI
- [x] MAUI Android
- [x] MAUI iOS
- [x] MAUI MacCatalyst

## Checklist

- [x] Self-review of changes
- [x] All changes work as expected on all affected platforms
- [x] There are no warnings related to changes
- [x] Code is commented and follows .NET conventions and standards
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] No unrelated changes have been made to any other code or project files
- [x] Screenshots are correct size and display in description tab (800 x 600 on Windows, MAUI mobile platforms use the MAUI Windows screenshot)

### CODE REVIEW

File: src/MAUI/Maui.Samples/Samples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.xaml.cs [hunk 1]
```diff
@@ -137,14 +146,18 @@ private void DisplayLabelsCheckBox_CheckedChanged(object sender, CheckedChangedE
             {
                 _clusteringFeatureReduction.LabelDefinitions.Clear();
             }
-            
         }
 
-        private void UpdateClusteringProperties(object sender, EventArgs e)
+        private void ClusterRadiusPicker_SelectedIndexChanged(object sender, EventArgs e)
```
Comment (duffh) [inline src/MAUI/Maui.Samples/Samples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.xaml.cs:151]: Do we need two event handlers here? 

Previously calling `UpdateClusteringProperties` covered both bases.

Also I think we should maintain our existing format of:

```
if(_layer != null)
{
    // do something
}
```

I don't think bracketless if statements are used in this format anywhere else.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.jpg
Comment (duffh) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.jpg:1]: Screenshot with orange checkbox background looks out of place.

File: src/MAUI/Maui.Samples/Samples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.xaml.cs (hunk 1 above)
Comment (williambohrmann3) [reply src/MAUI/Maui.Samples/Samples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.xaml.cs:151]: I kept running into null reference exceptions (specifically WPF) when I modified `UpdateClusteringProperties` to account for the swapping of the sliders with pickers. The issue is `MaxScalePicker.SelectedItem` is null when setting `ClusterRadiusPicker.SelectedItem`. Technically we might be able to null check on the picker selected items before trying to use them, however that introduces extra logic that can be avoided by having two event handlers. Furthermore, using two events allows each control to handle their own property setting separately, avoiding unnecessary property assignments. Happy to use curly braces for the if statements.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.jpg
Comment (williambohrmann3) [reply src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.jpg:1]: > Screenshot with orange checkbox background looks out of place.

WinUI uses the system theme color by default, which is orange for me and it looks like it is blue for you based on the previous screenshot. We want to display labels in the screenshot, so the checkbox background is filled with my system color.

Comment (duffh) [reply src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/AddClusteringFeatureReductionToAPointFeatureLayer/AddClusteringFeatureReductionToAPointFeatureLayer.jpg:1]: I think we should be consistent with our system color when taking capturing images of our samples as it looks strange having a mix of different colors in our screenshots, particularly when scrolling through the samples categories.

**Review by duffh — APPROVED** [review]

**Pull request merged**